### PR TITLE
Allow users to flip sprites and sprite sheets.

### PR DESCRIPTION
**Pull request by StarArawn**
A couple of people had brought up the lack of support for flipping sprites. I don't know if people will like trying to rotate or negatively scale sprites to flip them this offers a simple way of flipping them.

### CODE REVIEW

File: crates/bevy_sprite/src/render/sprite.vert [hunk 1]
```diff
@@ -17,8 +17,16 @@ layout(set = 2, binding = 1) uniform Sprite {
     vec2 Sprite_size;
 };
 
+layout(set = 1, binding = 1) uniform ColorMaterial_flip_horz {
```
Comment (cart) [inline crates/bevy_sprite/src/render/sprite.vert:20]: This is a valuable feature, but creating two new bindings for it seems a bit too much. I understand that the current "data driven" bindings encourage this design, but I don't want to go too far down that path. Ideally we either:
1. make it easier to consolidate these bindings into one
2. make these "ShaderDefs" instead (but that also has the problem of forcing more state changes / reducing batchability).

File: crates/bevy_sprite/src/render/sprite_sheet.vert [hunk 2]
```diff
@@ -32,6 +32,8 @@ layout(set = 2, binding = 0) uniform Transform {
 
 layout(set = 2, binding = 1) uniform TextureAtlasSprite {
     vec4 TextureAtlasSprite_color;
+    float flip_horz;
```
Comment (cart) [inline crates/bevy_sprite/src/render/sprite_sheet.vert:35]: wouldn't it make more sense to make these bools?

Comment (StarArawn) [reply crates/bevy_sprite/src/render/sprite_sheet.vert:35]: Yes, but currently render resources doesn't support booleans.

File: crates/bevy_sprite/src/render/sprite.vert (hunk 1 above)
Comment (StarArawn) [reply crates/bevy_sprite/src/render/sprite.vert:20]: Agreed. I want to make an issue for the bindings as it seems like this is something the average user is going to run into when creating a custom shader.

File: crates/bevy_sprite/src/render/sprite_sheet.vert (hunk 2 above)
Comment (cart) [reply crates/bevy_sprite/src/render/sprite_sheet.vert:35]: bawhaha oops

**Comment by cart** [comment]
I'm starting to think that disabling culling and allowing folks to flip via rotation might be the better short term solution. No additional bindings, no additional shaders, fits into the current 2d via 3d apis.

If we ever adopt a more 2d-focused api, "3d" rotations stop being possible. But thats likely farther down the road. Waiting for that gives us time to build the render features required to make this style of flipping "right".

Thoughts?

**Comment by StarArawn** [comment]
I think that makes sense. Closing this for now in favor of:
#241

**Pull request closed without merging**